Model/Config/Backend/Modules.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace RedChamps\CleanMenu\Model\Config\Backend;
+
+use Magento\Backend\Block\Menu;
+use Magento\Framework\App\Cache\Manager;
+use Magento\Framework\App\Cache\TypeListInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\Config\Value;
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\Framework\Model\Context;
+use Magento\Framework\Model\ResourceModel\AbstractResource;
+use Magento\Framework\Registry;
+
+class Modules extends Value
+{
+    /**
+     * @var Manager
+     */
+    private $cacheManager;
+
+    public function __construct(
+        Context $context,
+        Registry $registry,
+        ScopeConfigInterface $config,
+        TypeListInterface $cacheTypeList,
+        Manager $cacheManager,
+        AbstractResource $resource = null,
+        AbstractDb $resourceCollection = null,
+        array $data = []
+    ) {
+        $this->cacheManager = $cacheManager;
+        parent::__construct($context, $registry, $config, $cacheTypeList, $resource, $resourceCollection, $data);
+    }
+
+    public function afterSave()
```

Comment:
This method is not called at all when the setting is saved by checking "Use system config"

Reply:
I forget to remove "restore default", it does not make sense here, there is no default settings